met-api/src/met_api/resources/engagement.py

```diff
@@ -72,7 +72,7 @@ def get():
         try:
             args = request.args
             user_id = TokenInfo.get_id()
-            user_roles = TokenInfo.get_user_roles()
+            user_roles = TokenInfo.get_user_roles() if user_id else []
```

Comment:
I handled this in a previous PR..
Now tokeninfo.getuserrole handles this..so can be removed

Reply:
understood